### PR TITLE
gq: fix build with autotools

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

As suggested by @jmroot in: https://trac.macports.org/ticket/59054#comment:5

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by barracuda156** [comment]
Well, now it fails:
```
LC_ALL=C ../intltool-merge -x -u -c ../po/.intltool-merge-cache ../po gq-ldif.xml.in gq-ldif.xml
  LC_ALL=C ../intltool-merge -d -u -c ../po/.intltool-merge-cache ../po gq.desktop.in gq.desktop
  /bin/sh: ../intltool-merge: -w: bad interpreter: No such file or directory
  make[2]: *** [gq-ldif.xml] Error 126
```

**Comment by jmroot** [comment]
Oh, gq actually ships its own copies of the intltool executables, making this the first build system I've seen that actually uses the value of `INTLTOOL_PERL` after setting it. So ideally it would run the versions provided by the intltool port instead (assuming they are compatible). Maybe autoreconf with `-fvi` would help?

If that's not possible or too difficult, then gq would have to depend on intltool's dependencies so that it can guarantee the availability of the necessary perl modules for the perl version it is currently using.

**Comment by barracuda156** [comment]
@jmroot I can try using autotools. What is the criterion of success? FWIU it was building fine before, at least both CI bots and locally on my machines.

**Comment by jmroot** [comment]
The bug makes it fail to build when whatever `perl` configure happens to find doesn't have the XML::Parser module. It works accidentally if you have the perl5 port installed with the variant matching the perl version that intltool currently uses.

**Comment by jmroot** [comment]
This isn't theoretical by the way, you can see many, many tickets involving this failure mode in various ports by searching Trac for `XML::Parser`: https://trac.macports.org/search?q=XML%3A%3AParser&noquickjump=1&ticket=on

**Comment by barracuda156** [comment]
@jmroot I tried autotools, but it fails with:
```
configure.in:439: You should run autoupdate.
configure.in:5: error: possibly undefined macro: AM_ACLOCAL_INCLUDE
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
configure.in:67: error: possibly undefined macro: AS_PROG_OBJC
autoreconf: error: /opt/local/bin/autoconf failed with exit status: 1
```

To be explicit, I dropped intltool-perl patch and added this:
```
use_autoreconf  yes

depends_build-append \
                port:autoconf \
                port:automake \
                port:intltool \
                port:libtool \
                port:pkgconfig
```

**Comment by ryandesign** [comment]
> ```
> configure.in:5: error: possibly undefined macro: AM_ACLOCAL_INCLUDE
> ```

[A Google search of that error message suggests](https://wiki.mageia.org/en/Packaging_problems_and_solutions#error:_possibly_undefined_macro:_AM_ACLOCAL_INCLUDE) replacing `AM_ACLOCAL_INCLUDE` with `ACLOCAL_AMFLAGS`.

**Comment by barracuda156** [comment]
@jmroot @ryandesign Finally it worked. Needed one more fix to `configure.in`.

**Comment by jmroot** [comment]
You need to autoreconf so the changes to configure.in and Makefile.am are actually used.

**Comment by barracuda156** [comment]
> You need to autoreconf so the changes to configure.in and Makefile.am are actually used.

Sorry, my bad, autoreconf block somehow was missed (I have it locally, it vanished in rebasing). Gonna update now.

File: net/gq/Portfile [hunk 1]
```diff
@@ -21,7 +21,13 @@ checksums       rmd160  4e8bf1700f4d7aacd906559e01a11654f6ce0f90 \
...
 openssl.branch  1.0
 
-depends_build   port:intltool \
+use_autoreconf  yes
+
+depends_build-append \
+                port:autoconf \
+                port:automake \
+                port:intltool \
+                port:libtool \
```
Comment (jmroot) [inline net/gq/Portfile:30]: The autotools deps are added automatically by `use_autoreconf  yes`.

Comment (barracuda156) [reply net/gq/Portfile:30]: Do we still need `intltool` there?

Comment (jmroot) [reply net/gq/Portfile:30]: Yes.

Comment (barracuda156) [reply net/gq/Portfile]: Good now? Check passed.

**Review by jmroot — APPROVED** [review]
Thanks!

**Pull request merged**